diff --git a/clang/lib/Driver/ToolChain.cpp b/clang/lib/Driver/ToolChain.cpp
index bc70205a6c01..bfd34869913f 100644
--- a/clang/lib/Driver/ToolChain.cpp
+++ b/clang/lib/Driver/ToolChain.cpp
@@ -324,7 +324,7 @@ std::string ToolChain::getInputFilename(const InputInfo &Input) const {
 
 ToolChain::UnwindTableLevel
 ToolChain::getDefaultUnwindTableLevel(const ArgList &Args) const {
-  return UnwindTableLevel::None;
+  return UnwindTableLevel::Asynchronous;
 }
 
 Tool *ToolChain::getClang() const {
diff --git a/clang/lib/Driver/ToolChains/Gnu.cpp b/clang/lib/Driver/ToolChains/Gnu.cpp
index 814f3482a1b5..c492e45a2d0b 100644
--- a/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -2857,6 +2857,9 @@ void Generic_GCC::printVerboseInfo(raw_ostream &OS) const {
 
 ToolChain::UnwindTableLevel
 Generic_GCC::getDefaultUnwindTableLevel(const ArgList &Args) const {
+  if (getTriple().isMeeGo()) {
+    return UnwindTableLevel::Asynchronous;
+  }
   switch (getArch()) {
   case llvm::Triple::aarch64:
   case llvm::Triple::ppc:
